### PR TITLE
Issue #1311: Improve EL handling for the identity stores.

### LINKED ISSUE #1311: Update EL handling in Database and LDAP identity stores
The EL handling is not handling possible NPEs and in particular does not properly handle the hashAlgorithmParameters for the DatabaseIdentityStore.

Additionally, the LdapIdentityStore.getGroupsByMember() method is not using groupNameAttribute to construct the group name.

### PR DESCRIPTION
**Pull request by jvanhill**
fixes #1311

### CODE REVIEW

**Comment by jvanhill** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.security.javaeesec_fat

**Comment by jvanhill** [comment]
Changes to fix setting of group name to the value of groupsMemberAttribute in LdapIdentityStore.getGroupsByMember() broke some tests. Updated the tests and issuing a rebuild.

#build
#spawn.fullfat.buckets=com.ibm.ws.security.javaeesec_fat

File: dev/com.ibm.ws.security.javaeesec/resources/com/ibm/ws/security/javaeesec/internal/resources/JavaEESecMessages.nlsprops [hunk 1]
```diff
@@ -42,3 +42,7 @@ JAVAEESEC_WARNING_NO_REALM_NAME.useraction=If the realm name needs to be set, pl
 JAVAEESEC_ERROR_MULTIPLE_HTTPAUTHMECHS=CWWKS1915E: The deployment for the {0} module in the {1} application failed because multiple HttpAuthenticationMechanism implementation are found. The list of classes is {2}. Most likely, this is an application packaging issue. Make sure that there is only one HttpAuthenticationMechanism implementation in each module.
 JAVAEESEC_ERROR_MULTIPLE_HTTPAUTHMECHS.explanation=The application deployment failed because more than one HttpAuthenticationMechanism implementation are found. This condition is not allowed.
 JAVAEESEC_ERROR_MULTIPLE_HTTPAUTHMECHS.useraction=Make sure that there is only one HttpAuthenticationMechanism implementation for each module in the application.
+
+JAVAEESEC_WARNING_IDSTORE_CONFIG=CWWKS1916W: There was an error resolving the ''{0}'' configuration for the identity store. Ensure any EL expressions are resolvable. The value will be defaulted to ''{1}''.
```
Comment (sishida) [inline dev/com.ibm.ws.security.javaeesec/resources/com/ibm/ws/security/javaeesec/internal/resources/JavaEESecMessages.nlsprops]: +JAVAEESEC_WARNING_IDSTORE_CONFIG=CWWKS1916W: An error occurred when resolving the ''{0}'' configuration for the identity store. Ensure any EL expressions are resolvable. The value defaults to ''{1}''.

File: dev/com.ibm.ws.security.javaeesec/resources/com/ibm/ws/security/javaeesec/internal/resources/JavaEESecMessages.nlsprops [hunk 2]
```diff
@@ -42,3 +42,7 @@ JAVAEESEC_WARNING_NO_REALM_NAME.useraction=If the realm name needs to be set, pl
...
+JAVAEESEC_WARNING_IDSTORE_CONFIG.explanation=The identity store configuration could not be resolved to a valid value.
```
Comment (sishida) [inline dev/com.ibm.ws.security.javaeesec/resources/com/ibm/ws/security/javaeesec/internal/resources/JavaEESecMessages.nlsprops]: +JAVAEESEC_WARNING_IDSTORE_CONFIG.explanation=The identity store configuration cannot be resolved to a valid value.

Note: Can we tell the user how they can resolve the issue, such as adding a sentence that says how they can get valid values?

File: dev/com.ibm.ws.security.javaeesec/resources/com/ibm/ws/security/javaeesec/internal/resources/JavaEESecMessages.nlsprops [hunk 3]
```diff
@@ -42,3 +42,7 @@ JAVAEESEC_WARNING_NO_REALM_NAME.useraction=If the realm name needs to be set, pl
...
+JAVAEESEC_WARNING_IDSTORE_CONFIG.useraction=Make sure that the annotation contains a valid configuration value. If using EL expressions, ensure they are valid and that any beans are resolvable.
```
Comment (sishida) [inline dev/com.ibm.ws.security.javaeesec/resources/com/ibm/ws/security/javaeesec/internal/resources/JavaEESecMessages.nlsprops]: +JAVAEESEC_WARNING_IDSTORE_CONFIG.useraction=Make sure that the annotation contains a valid configuration value. If you use EL expressions, ensure that they are valid and that any beans are resolvable.

File: dev/com.ibm.ws.security.javaeesec/resources/com/ibm/ws/security/javaeesec/internal/resources/JavaEESecMessages.nlsprops (hunk 2 above)
Comment (sishida) [reply dev/com.ibm.ws.security.javaeesec/resources/com/ibm/ws/security/javaeesec/internal/resources/JavaEESecMessages.nlsprops]: Update:
Ensure that the EL expression and the result are valid and ensure any CDI beans that are used in expressions are resolvable.

**Comment by sishida** [comment]
Discussed the messages with Jesse over Sametime. Finished reviewing the changes there. :)

**Pull request merged**